### ISSUE TITLE
HTTP link is not clickable when message text longer than one screen

### ISSUE THREAD

**Issue by fracting** [oxen-io/session-android#1119]
**Code of conduct**

- [x] I have read and agree to adhere to the [Code of Conduct](https://github.com/oxen-io/session-android/blob/master/CODE_OF_CONDUCT.md).

**Describe the bug**

Fresh installed Session on Android simulator (Other users report the same bug on real Android devices as well)

Send or receive a long text with some links, if the message is longer than the screen, the the links in the bottom are uncliclable.

**To reproduce**

Text to test
```
XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

https://oxen.io
```

**Screenshots or logs**

If applicable, add screenshots or logs to help explain your problem.

<img width="301" alt="Screen Shot 2023-02-15 at 6 28 37 pm" src="https://user-images.githubusercontent.com/194342/218961912-d7ada489-4388-416c-b354-00ddc8b74915.png">

**Smartphone (please complete the following information):**

 - Device: [e.g. Pixel 6]
 - OS: [e.g. Android 13]
 - Version of Session or latest commit hash
 Custom built Session Android 395ada62fffa7031244cd0a6f144a20e7e824562

**Additional context**

Same bug reproducible on 1-1 chat, closed group, community and "Note to Self"

**Comment by fracting** [comment]
This test case shows a bit more interesting behavior:

When clicking on `https://oxen.io/1` to `https://oxen.io/6`, nothing happens
When clicking on `https://oxen.io/7` to `https://oxen.io/9`, ModalURLSpan of `https://oxen.io/1` to `https://oxen.io/3` is displayed.

```
XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX

XXXXXXXXXXXXXX
https://oxen.io/1
https://oxen.io/2
https://oxen.io/3
https://oxen.io/4
https://oxen.io/5
https://oxen.io/6
https://oxen.io/7
https://oxen.io/8
https://oxen.io/9
```

**Comment by fracting** [comment]
Hi @bemusementpark while you are working on other scroll / visibility related UI glitch bugs, would you mind take a look at this one to see if they are related?

Thanks!

**Comment by bemusementpark** [comment]
Hey @fracting that's not directly related, but it shouldn't be a difficult fix, it looks like the issue occurs whenever the top of the view is off-screen.

More specifically the `VisibleMessageView#y` is not being incorporated into the link hitTest when `y` is negative.

**Comment by bemusementpark** [comment]
https://github.com/oxen-io/session-android/pull/1184

**Comment by fracting** [comment]
Confirming fixed in dev branch, I'll retest and close this issue when the next official release is published.

**Comment by fracting** [comment]
Fixed in 1.16.9. Close.
